content/articles/2017-12-14-styler-1-0-0.Rmd

```diff
@@ -0,0 +1,230 @@
+
+Also, with `scope >= "line_breaks"`, `strict = TRUE` will break the lines 
+in the following call, whreas `strict = FALSE` will leave them as is.
+```{r}
+style_text(
+  "do_a_long_and_complicated_fun_cal('which', has, way, to, 
+                              'and longer then lorem ipsum in its full length'
+                              )"
+)
+```
+
+
+In addition, the `strict` option does also adds braces (if `scope = "tokens"`) to conditional statements if they are multi-line.
+```{r}
+style_text(
+  "if (set_q(x))
+    3 else 
+    5",
+  strict = TRUE
+)
+```
+
+
+### `math_token_spacing`
+
+styler can identify and handle unary operators and other math tokens:
+
+```{styler}
+1++1-1-1/2
+```
+
+This is tidyverse style. However, styler offers very granular control for 
+math token spacing. Assuming you like spacing around `+` and `-`, but not 
+around `/` and `*` and `^`. This can be achieved as follows:
+```{r}
+style_text(
+  "1++1/2*2^2", 
+  math_token_spacing = specify_math_token_spacing(zero = c("'/'", "'*'", "'^'"))
+)
+```
+
+### `reindention`
+
+You can set the indention of lines that match a certain regular expression manually. For example, if you don't want comments starting with `###` to be indented, you can formulate an unindention rule:
+```{r}
+style_text(
+    "a <- function() {
+    ### not to be indented
+    # indent normally
+    33
+    }",
+  reindention = specify_reindention(regex_pattern = "^###", indention = 0)
+)
+```
+
+
+These are options that allow flexible reformatting. You can even go further, as described in [this](http://styler.r-lib.org/articles/customizing_styler.html) vignette.
+
+# A few more examples
+
+Now that you know that the rules can be applied in a flexible fashion, let's focus on `scope = "tokens"` and `strict = TRUE` and illustrate how specific rulres of the tidyverse style guide are implemented. 
+
+Function declarations are indented if multi-line:
+```{styler}
+my_fun <- function(x, 
+y, 
+z) {
+  just(z)
+}
+```
+
+styler can also format complicated expressions that involve line breaking and 
+indention based on both brace expressions and operators:
+
+```{styler}
+if (x >3) {stop("this is an error")} else {
+c(there_are_fairly_long,
+1 / 33 * 
+2 * long_long_variable_names)%>% k(
+
+) }
+```
+
+Lines are broken after `(` if a function call spans multiple lines:
+
+```{styler}
+do_a_long_and_complicated_fun_cal("which", has, way, to, 
+                              "and longer then lorem ipsum in its full length"
+                              )
+```
+
+styler replaces `=` with `<-` for assignment if necessary.
+```{r}
+style_text(
+  "
+  one = 3
+  data_frame(a = 3)
+  "
+)
+```
+
+It also handles single quotes according to the tidyverse style guide.
+```{r}
+'one string'
+"one string in a 'string'"
+
+```
+
+Braces are added to function calls in pipes:
+
+```{styler}
+a %>%
+  b %>%
+  c
+```
+
+tidyeval syntax is handled properly:
+```{styler}
+mtcars %>%
+  group_by(!!!my_vars)
+```
```

Comment:
Not sure how to conclude the article. Other article did not seem to have a wrap-up section.

Reply:
Maybe conclude with a CTA, like "grab styler from CRAN, run it on your own package, tell us what you think"?

Reply:
Does CTA stand for 'call to action'? So I can just add a last sentence without a header...

Reply:
Yes. Not sure if we need a header either.